pkg/scheduler/factory/factory.go

```diff
@@ -822,24 +822,6 @@ func (c *configFactory) CreateFromConfig(policy schedulerapi.Policy) (*Config, e
 	return c.CreateFromKeys(predicateKeys, priorityKeys, extenders)
 }
 
-// getBinderFunc returns an func which returns an extender that supports bind or a default binder based on the given pod.
-func (c *configFactory) getBinderFunc(extenders []algorithm.SchedulerExtender) func(pod *v1.Pod) Binder {
-	var extenderBinder algorithm.SchedulerExtender
-	for i := range extenders {
-		if extenders[i].IsBinder() {
-			extenderBinder = extenders[i]
-			break
-		}
-	}
-	defaultBinder := &binder{c.client}
-	return func(pod *v1.Pod) Binder {
-		if extenderBinder != nil && extenderBinder.IsInterested(pod) {
-			return extenderBinder
-		}
-		return defaultBinder
-	}
-}
-
```

Comment:
@misterikkit just for curiisity why did this function not needed to be a method . it was using c.client from the object ? Any other reason ?

Reply:
@krmayankk Yeah, the purpose here is to remove the dependency on the `configFactory` type. At the end of phase :panda_face: we should be able to construct a `Scheduler` instance without ever creating a `configFactory` object.

Also, since the *only* member of `configFactory` being accessed was `c.client` (out of many fields) this helps to clarify the scope of what `GetBinderFunc` actually needs.

Reply:
makes sense. i think the general thinking is that if its using one or two of the fields of configFactory, then it probably can be a function, agreed with not depending on configFactory at all if it directly takes the client as input